### PR TITLE
Add performance test for rich console in various scenarios

### PR DESCRIPTION
**Pull request by bmuschko**
### Context

Verify performance for various scenarios.

### CODE REVIEW

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+class RichConsolePerformanceTest extends AbstractCrossVersionPerformanceTest {
+
+    private static final List<String> CLEAN_ASSEMBLE_TASKS = ['clean', 'assemble']
+
+    def setup() {
+        runner.args << '--console=rich'
+    }
+
+    @Unroll
+    def "can execute on #testProject with rich console"() {
```
Comment (oehme) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy]: This is not an integration test and should not start with "can". Please see all the other perf tests for how to name them.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        given:
+        runner.testProject = testProject
+        runner.tasksToRun = tasks
+        runner.gradleOpts = ["-Xms${testProject.daemonMemory}", "-Xmx${testProject.daemonMemory}"]
+        runner.targetVersions = ['3.5-20170221000043+0000']
```
Comment (oehme) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy]: Unnecessary.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        when:
+        def result = runner.run()
+
+        then:
+        result.assertCurrentVersionHasNotRegressed()
+
+        where:
```
Comment (oehme) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy]: These scenarios will take a very long time if run with the usual 10 warmups and 30 iterations. We should either use different scenarios or reduce the number of warmups/runs. See the other clean assemble tests.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy (hunk 1 above)
Comment (bmuschko) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy]: done

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy (hunk 2 above)
Comment (bmuschko) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy]: done

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy (hunk 3 above)
Comment (bmuschko) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy]: done

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+class RichConsolePerformanceTest extends AbstractCrossVersionPerformanceTest {
+
+    private static final List<String> CLEAN_ASSEMBLE_TASKS = ['clean', 'assemble']
+
+    def setup() {
+        runner.args << '--console=rich'
+    }
+
+    @Unroll
+    def "execute #testProject with rich console"() {
```
Comment (oehme) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy]: #taskNames on #testProject with rich console

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy (hunk 3 above)
Comment (oehme) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy]: We haven't actually reduced the number of warmups/runs. See the other clean assemble tests for reference how many is reasonable.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy (hunk 4 above)
Comment (bmuschko) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/RichConsolePerformanceTest.groovy]: I will fix it though other test cases use a verb in the beginning as well e.g.

- create many empty projects
- merge exclude rules
- get IDE model on #testProject for Android Studio

**Comment by bmuschko** [comment]
@oehme Addressed your comment.

**Pull request merged**